src/jbuf/jbuf.c

```diff
@@ -190,10 +190,12 @@ static uint32_t calc_ptime(struct jbuf *jb)
 	if (lt && lt->prev) {
 		ft  = lt->data;
 		ftt = lt->prev->data;
-		seqd = ft->hdr.seq - ftt->hdr.seq;
-		if (ftt && ft && seqd > 0)
-			ptime = (ft->hdr.ts - ftt->hdr.ts) / (8 * seqd);
-
+		if (ftt && ft) {
+			seqd = ft->hdr.seq - ftt->hdr.seq;
+			if (seqd > 0)
+				ptime = (ft->hdr.ts - ftt->hdr.ts)
+					/ (8 * seqd);
+		}
 	}
```

Comment:
I think we can optimize the if nesting a bit, like this:

```c
if (!ftt || !ft)
	return ptime;
```

Reply:
I agree with you, your code is better